hpy/devel/include/common/hpytype.h

```diff
@@ -48,6 +48,20 @@ typedef struct {
 #define _Py_TPFLAGS_HEAPTYPE (1UL << 9)
 #define _Py_TPFLAGS_HAVE_VERSION_TAG (1UL << 18)
 
+
+/* HPy_TPFLAGS_LEGACY is set automatically on legacy types.
+
+   A type is a legacy type if legacy_headersize is greater than zero. This
+   is typically true if HPyObject_HEAD is included in the type struct
+   and the type sets legacy_* slots or methods.
+
+   Note on the choice of bit 8: Bit 8 looks likely to be the last free TPFLAG
+   bit that C Python will allocate. Bits 0 to 8 were dropped in Python 3.0 and
+   are being re-allocated slowly from 0 towards 8. As of 3.10, only bit 0 has
+   been re-allocated.
+*/
+#define HPy_TPFLAGS_LEGACY (1UL << 8)
+
```

Comment:
Reversing the flag sounds good, so I'll do that now.

Currently `.legacy` overrides the `flags` setting entirely, so setting `.flags` to include `HPy_TPFLAGS_LEGACY` has no effect, and I'll keep that for `HPy_TPFLAGS_PURE`.

I think we should keep the flag public even though it should never be used directly because none of the public flags should ever have the same value. It would be nice to avoid having to use this flag entirely, but I don't see how we could sanely store this bit of information on the `PyType_Spec` other than as a flag.

Reply:
I agree that we have to use the flag, but adding a sanity check to ensure that you don't have conflicting values for `.legacy` and `HPy_TPFLAGS_LEGACY` would be nice. Anyway, as I said I am fine to discuss it in a follow-up PR